repair/repair_task.hh

```diff
@@ -193,6 +193,14 @@ public:
     virtual tasks::is_internal is_internal() const noexcept override {
         return tasks::is_internal::yes;
     }
+
+    virtual future<tasks::task_manager::task::progress> get_progress() const override {
+        auto state = _status.state;
+        co_return tasks::task_manager::task::progress{
+            .completed = state == tasks::task_manager::task_state::done || state == tasks::task_manager::task_state::failed,
```

Comment:
don't we have a method to check the task_state for completion?
If not, let's add one since the logic is generic and can very well be used by other task types.

Reply:
Maybe this could even be the generic implementation of this virtual function in the base class.